security/pkg/nodeagent/sds/secretmanager.go

```diff
@@ -18,14 +18,17 @@ import "time"
 
 // SecretManager defines secrets management interface which is used by SDS.
 type SecretManager interface {
-	GetSecret(proxyID, token string) (*SecretItem, error)
+	GetSecret(proxyID, serviceAccount, token string) (*SecretItem, error)
 }
 
 // SecretItem is the cached item in in-memory secret store.
 type SecretItem struct {
 	CertificateChain []byte
 	PrivateKey       []byte
 
+	// ServiceAccount passed from envoy, in spiffe format.
+	ServiceAccount string
```

Comment:
The service account in GCP is `foo@iam.gserviceaccount.com`, in kubernetes is like `foo`, and their corresponding spiffe id is `spiffe://accounts.google.com/foo` and `spiffe://cluster.local/ns/namespace/sa/foo`. So I think to be precise this should be `SpiffeID`?

Reply:
renamed.